akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala

```diff
@@ -422,9 +441,11 @@ class ForkJoinExecutorConfigurator(config: Config, prerequisites: DispatcherPrer
   }
 
   class ForkJoinExecutorServiceFactory(val threadFactory: ForkJoinPool.ForkJoinWorkerThreadFactory,
-                                       val parallelism: Int) extends ExecutorServiceFactory {
-    def createExecutorService: ExecutorService = new AkkaForkJoinPool(parallelism, threadFactory, MonitorableThreadFactory.doNothing)
+                                       val parallelism: Int,
+                                       val asyncMode: Boolean) extends ExecutorServiceFactory {
```

Comment:
That's the binary compat failure. Technically it's not akka-private so people may have used it.
We could add an additional constructor here to be binary compatible - with the default value as before-this-patch.

// cc @rkuhn